Gap filling of load time series does not work for the default configuration

In the current default configuration, load time series from 2016 is used for all countries in Europe. We have load data of Albania (ALB) only starting from 2017 and thus, `euro-calliope 1.0` used to use 2017 data for Albania.

The current gap filling method should do the same (as I understand it), but it does not do it. Therefore, `develop` is currently broken with the following error message: 

``AssertionError: Not enough load data for country ALB``

--- a/scripts/national_load.py
+++ b/scripts/national_load.py
@@ -58,7 +58,7 @@ def _get_index_of_missing_data(series, model_year):
     return series[(series < 0) | series.isna()][str(model_year)].index
 
 
-def _handle_feb_29th(this_year, next_avail_year, missing_timesteps):
+def _ignore_feb_29th(this_year, next_avail_year, missing_timesteps):
     """ Ignore February 29th if next available year doesn't have that data available """
     if (pd.Period(freq='Y', year=this_year).is_leap_year and
             not pd.Period(freq='Y', year=next_avail_year).is_leap_year and
@@ -78,7 +78,7 @@ def fill_missing_data_in_country(country_series, model_year, acceptable_gap_hour
     # We're OK with some gap, as defined by 'acceptable_gap_hours'
     years_with_data = country_series.dropna().index.year.drop(model_year, errors="ignore").unique()
 
-    def _get_year_order(years):
+    def _give_older_years_lower_priority(years):
         year_order = []
         for year in years:
             if year - model_year < 0:
@@ -86,23 +86,22 @@ def fill_missing_data_in_country(country_series, model_year, acceptable_gap_hour
             else:
                 year_order.append(year - model_year)
         return year_order
-    preferred_order_of_years_with_data = list(years_with_data.sort_values(key=_get_year_order, ascending=False))
+    preferred_order_of_years_with_data = list(years_with_data.sort_values(key=_give_older_years_lower_priority, ascending=False))
 
-    while country_series.loc[all_missing_timesteps].isnull().sum() > acceptable_gap_hours:
+    _missing_timesteps = all_missing_timesteps.copy()
+    while len(_missing_timesteps) > acceptable_gap_hours:
         if preferred_order_of_years_with_data == []:
             break
 
         next_avail_year = preferred_order_of_years_with_data.pop(0)
         _missing_timesteps = _get_index_of_missing_data(country_series, model_year)
-        _missing_timesteps = _handle_feb_29th(model_year, next_avail_year, _missing_timesteps)
+        _missing_timesteps = _ignore_feb_29th(model_year, next_avail_year, _missing_timesteps)
         new_data = country_series.reindex(_missing_timesteps.map(lambda dt: dt.replace(year=next_avail_year)))
         new_data.index = _missing_timesteps
         country_series.update(new_data)
         fill_years += [str(next_avail_year)]
 
-    assert columns_with_missing_data_in_model_year(
-        country_series.to_frame(), model_year, acceptable_gap_hours
-    ).empty, f"Not enough load data for country {country_series.name}"
+    assert country_series.loc[str(model_year)].isna().sum() <= acceptable_gap_hours, f"Not enough load data for country {country_series.name}"
 
     return country_series.loc[all_missing_timesteps], len(all_missing_timesteps), fill_years
 

--- a/scripts/national_load.py
+++ b/scripts/national_load.py
@@ -1,5 +1,5 @@
 """Preprocess national electricity load time series."""
-from datetime import datetime, timezone
+import calendar
 
 import pandas as pd
 import numpy as np
@@ -60,9 +60,10 @@ def _get_index_of_missing_data(series, model_year):
 
 def _ignore_feb_29th(this_year, next_avail_year, missing_timesteps):
     """ Ignore February 29th if next available year doesn't have that data available """
-    if (pd.Period(freq='Y', year=this_year).is_leap_year and
-            not pd.Period(freq='Y', year=next_avail_year).is_leap_year and
-            pd.to_datetime(f'{this_year}-02-29').date() in missing_timesteps.date):
+    if (calendar.isleap(this_year)
+        and not calendar.isleap(next_avail_year)
+        and pd.to_datetime(f'{this_year}-02-29').date() in missing_timesteps.date
+    ):
         return missing_timesteps[
             missing_timesteps.date != pd.to_datetime(f'{this_year}-02-29').date()
         ]
@@ -89,10 +90,7 @@ def fill_missing_data_in_country(country_series, model_year, acceptable_gap_hour
     preferred_order_of_years_with_data = list(years_with_data.sort_values(key=_give_older_years_lower_priority, ascending=False))
 
     _missing_timesteps = all_missing_timesteps.copy()
-    while len(_missing_timesteps) > acceptable_gap_hours:
-        if preferred_order_of_years_with_data == []:
-            break
-
+    while len(_missing_timesteps) > acceptable_gap_hours and preferred_order_of_years_with_data != []:
         next_avail_year = preferred_order_of_years_with_data.pop(0)
         _missing_timesteps = _get_index_of_missing_data(country_series, model_year)
         _missing_timesteps = _ignore_feb_29th(model_year, next_avail_year, _missing_timesteps)

